customfield.php

```diff
@@ -33,21 +33,31 @@
 
 admin_externalpage_setup('mod_cms/managetypes');
 
+$url = new moodle_url('/mod/cms/customfield.php', ['itemid' => $itemid]);
 $PAGE->set_context(context_system::instance());
-$PAGE->set_url(new moodle_url('/mod/cms/customfield.php', ['itemid' => $itemid]));
+$PAGE->set_url($url);
 
 $cfhandler = cmsfield_handler::create($itemid);
 
 $cmstype = new cms_types($itemid);
 
 $returnurl = new moodle_url('/mod/cms/managetypes.php');
 
+$pageheading = get_string('customfield_manage_heading', 'mod_cms', $cmstype->get('name'));
+
+// Need to add extra breadcrumbs when URL does not exactly match admin setting page URL.
+foreach (['modsettings', 'modcmsfolder', 'mod_cms/managetypes'] as $label) {
+    if ($node = $PAGE->settingsnav->find($label, \navigation_node::TYPE_SETTING)) {
```

Comment:
`use navigation_node;`
```suggestion
    if ($node = $PAGE->settingsnav->find($label, navigation_node::TYPE_SETTING)) {
```